linode/manager
# feat: Manual Backups for Managed Databases

## Description 📝

**What does this PR do?**

Enables managed database users to trigger backups on demand.  This will hit the `POST` endpoint with a custom label, defined here in the public API: https://www.linode.com/docs/api/databases/#managed-mysql-database-backup-snapshot-create

Since we have a limit on the number of manual backups, this PR also adds the ability to delete some in order to make room for more.  Deletion works for either automatic or manual backups.  It hits the `DELETE` endpoint defined here: https://www.linode.com/docs/api/databases/#managed-mongodb-database-backup-delete

The confirmation dialogs have been changed such that:
* The formatted date should be entered to restore an automatic or manual backup
* The formatted date should be entered to delete a manual backup
* Deleting automatic backups is not supported.  Deleting them does not count against the 3 manual backup limit.

Thoughts on this? ☝️ 

## Preview 📷

https://user-images.githubusercontent.com/5467060/213305403-950a5935-f837-4851-b6c7-d2e36d565add.mov

## How to test 🧪

**What are the steps to reproduce the issue or verify the changes?**

Head to the managed databases using the left sidebar.  Create a new managed database (or use an existing one).  Select your database, then hit the `Backups` tab.  From here, you should see a `Manual Backup` section, where you'll be able to click the blue `Start Backup` button.  Enter a custom label using only alphanumeric characters and no spaces, then click `Backup Database`.  Observe that the correct endpoint is hit successfully.

**How do I run relevant unit or e2e tests?**

`yarn test` (for now)
